docs/how-to/testing.rst

```diff
@@ -0,0 +1,101 @@
+=============
+Running tests
+=============
+
+``MicroOVN`` has two types of tests, linter checks and functional tests.
+
+Linter checks
+-------------
+
+Linting is currently applied only to the tests themselves, not the main
+codebase. Prerequisite for running linters are:
+
+* ``make``
+* ``shellcheck``
+
+To perform linting, run following command:
+
+.. code-block:: none
+
+   make check-lint
+
+Functional tests
+----------------
+
+These tests build ``MicroOVN`` snap and use it to deploy ``OVN`` cluster in
+LXD containers. This cluster is then used for running functional test suites.
+
+Preparing environment
+~~~~~~~~~~~~~~~~~~~~~
+
+There is no need to run tests in dedicated VMs or in isolated environments as
+all functional tests run inside containers and no changes are made to the host
+running them.
+
+Functional test need to build ``MicroOVN`` snap and run multiple containers via
+LXD, so make sure you first complete steps and requirements for
+:doc:`building MicroOVN from source <building>`.
+
+Final step is to ensure that you have required `Bash Automated Testing System
+(BATS)`_. Due to reliance on latest features, ``MicroOVN`` uses ``BATS``
+directly from the source. If you cloned ``MicroOVN`` repository with submodules
+(using ``--recurse-submodules`` flag), you are all set and you will have
+following **non-empty** directories:
+
+* ``.bats/bats-assert/``
+* ``.bats/bats-core/``
+* ``.bats/bats-support/``
+
+If they are empty, you can fetch the submodules with:
+
+.. code-block:: none
+
+   git submodule update --init --recursive
+
+Running functional tests
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once you have your environment set up, running tests is just a matter of
+invoking right ``make`` target. To run all available test suites you can
```

Comment:
the appropriate ``make`` target. To run all the available test suites, use the ``check-system`` make target: